File: lucette/src/main/java/battle/csn/lucette/machine/states/PlayState.java
package battle.csn.lucette.machine.states;

import java.util.ArrayList;
import java.util.List;

import org.apache.log4j.Logger;

import battle.csn.lucette.game.GameEngine;
import battle.csn.lucette.game.Moteur;
import battle.csn.lucette.game.board.IBoard;
import battle.csn.lucette.game.structure.Move;
import battle.csn.lucette.machine.Chain;
import battle.csn.lucette.machine.FightStateMachine;
import battle.csn.lucette.machine.states.framework.AbstractIntermediateState;
import battle.csn.lucette.service.RestClientAPI;


public class PlayState extends AbstractIntermediateState
{
    private static final String OUI = "OK";
    private static final String KO = "KO";
    private static final String PTT = "PTT";
    private static final String GAGNE = "GAGNE";

    @Override
    public void pull( Chain chain )
    {
        FightStateMachine stateMachine = chain.getStateMachine( );

        // TODO : here the magic happend.

        String strBoard = new RestClientAPI( ).getBoard( chain.getStateMachine( ).getGame( ) );

        Moteur moteur = new Moteur( "" );
        IBoard plateau = moteur.getPlateau( );
        plateau.updateCases( strBoard );

        List<String> idEquipes = new ArrayList<>( );
        idEquipes.add( chain.getStateMachine( ).getIdEquipe( ) );
        idEquipes.add( "TODO adv" );
        plateau.setIdEquipes( idEquipes );

        Move move = new GameEngine( ).getMove( plateau, chain.getStateMachine( ).getIdEquipe( ) );
        List<Integer> positions = move.getPositions( );

        String result = new RestClientAPI( ).play( stateMachine.getGame( ), stateMachine.getIdEquipe( ), positions );

        switch ( result )
        {
        case OUI:
            chain.setState( new InitialState( ) );
            break;
        case KO:
            chain.setState( new ErrorState( "We did a forbidden move here :/" ) );
            break;
        case PTT:
            System.out.println( "Why the f**k did we played now ???" );
            chain.setState( new InitialState( ) );
            break;
        case GAGNE:
            chain.setState( new GagneState( ) );
            break;
        default:
            chain.setState( new ErrorState( "I don't know what to do with this" ) );
            break;
        }

    }
}

File: trunk/lucette2/src/main/scala/battle/csn/lucette2/engine/states/PlayState.scala
package battle.csn.lucette2.engine.states

import battle.csn.lucette2.engine.states.framework.AbstractIntermediateState
import battle.csn.lucette2.engine.Chain
import battle.csn.lucette2.network.RestClient
import battle.csn.lucette2.engine.FightStateMachine
import battle.csn.lucette2.engine.Engine
import scala.collection.mutable.MutableList
import battle.csn.lucette2.game.structure.Move

class PlayState extends AbstractIntermediateState {

  def pull(chain: Chain) {
    var stateMachine: FightStateMachine = chain.stateMachine

    var strBoard = new RestClient().getBoard(chain.stateMachine.game)

    var engine = new Engine();
    var plateau = engine.plateau
    plateau.updateBoard(strBoard);

    var idEquipes = MutableList[String]()
    idEquipes += chain.stateMachine.idEquipe
    idEquipes += "TODO adv"
    //TODO implement new solution
    //plateau.( idEquipes );

    //TODO implement
    //var move = new GameEngine( ).getMove( plateau, chain.getStateMachine( ).getIdEquipe( ) );
    var move = Move()

    var result = new RestClient().play(stateMachine.game, stateMachine.idEquipe, move.positions);

    result match {
      case OUI =>
        chain.current = new InitialState()
      case KO =>
        chain.current = new ErrorState("We did a forbidden move here :/")
      case PTT =>
        System.out.println("Why the f**k did we played now ???")
        chain.current = new InitialState()
      case GAGNE =>
        chain.current = new WinState()
      case default =>
        chain.current = new ErrorState("I don't know what to do with this")
    }
  }

}